lib/chef/resource/swap_file.rb

```diff
@@ -0,0 +1,217 @@
+#
+# Copyright 2012-2018, Seth Vargo
+# Copyright 2017-2018, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class SwapFile < Chef::Resource
+      resource_name :swap_file
+      provides :swap_file
+
+      description "Use the swap_file resource to create or delete swap files on Linux systems."
+      introduced "14.0"
+
+      property :path, String,
+               description: "The path to put the swap file on the system.",
+               name_property: true
+
+      property :size, Integer,
+               description: "The size (in MBs) for the swap file."
+
+      property :persist, [TrueClass, FalseClass],
+               description: "Persist the swapon.",
+               default: false
+
+      property :timeout, Integer,
+               description: "Timeout for dd/fallocate.",
+               default: 600
+
+      property :swappiness, Integer,
+               description: "The swappiness value to set on the system."
+
+      action :create do
+        description "Create a swapfile."
+
+        if swap_enabled?
+          Chef::Log.debug("#{new_resource} already created - nothing to do")
+        else
+          begin
+            Chef::Log.info "starting first create: #{node['virtualization']['system']}"
+            do_create(swap_creation_command)
+          rescue Mixlib::ShellOut::ShellCommandFailed => e
+            Chef::Log.warn("#{new_resource} Rescuing failed swapfile creation for #{new_resource.path}")
+            Chef::Log.debug("#{new_resource} Exception when creating swapfile #{new_resource.path}: #{e}")
+            do_create(dd_command)
+          end
+        end
+        if new_resource.swappiness
+          include_recipe "sysctl::default"
+
+          sysctl_param "vm.swappiness" do
```

Comment:
Should use `sysctl` as the new primary name.

Reply:
Done and the include is gone too